controllers/validation/accountpool_validation_controller.go

```diff
@@ -0,0 +1,198 @@
+	if err != nil {
+		logs.Info("Could not retrieve feature flag 'feature.accountpool_validation' - accountpool validation is disabled")
+	} else {
+		accountPoolValidationEnabled = enabled
+	}
+	logs.Info("Is accountpool_validation enabled?", "enabled", accountPoolValidationEnabled)
+
+	reqLogger.Info("Checking ConfigMap for ServiceQuotas")
+	// check if accountpool has servicequota defined in configmap
+	reginalServiceQuotas, err := utils.ParseAccountPoolData(reqLogger, currentAccountPool.Name, r.Client)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	reqLogger.Info("Updating Account ServiceQuotas")
+	_, err = r.checkAccountServiceQuota(reqLogger, currentAccountPool.Name, reginalServiceQuotas, accountPoolValidationEnabled)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	return utils.DoNotRequeue()
+}
+
+func (r *AccountPoolValidationReconciler) accountSpecUpdate(reqLogger logr.Logger, account *awsv1alpha1.Account) error {
+	err := r.Client.Update(context.TODO(), account)
+	if err != nil {
+		reqLogger.Error(err, fmt.Sprintf("Account spec update for %s failed", account.Name))
+	}
+	return err
+}
+
+func (r *AccountPoolValidationReconciler) accountStatusUpdate(reqLogger logr.Logger, account *awsv1alpha1.Account) error {
+	err := r.Client.Status().Update(context.TODO(), account)
+	if err != nil {
+		reqLogger.Error(err, fmt.Sprintf("Account status update for %s failed", account.Name))
+	}
+	return err
+}
+
+func (r *AccountPoolValidationReconciler) getAccountPoolAccounts(accountPoolName string) ([]awsv1alpha1.Account, error) {
+	//Get the number of actual unclaimed AWS accounts in the pool
+	accountList := &awsv1alpha1.AccountList{}
+	listOpts := []client.ListOption{
+		client.InNamespace(awsv1alpha1.AccountCrNamespace),
+	}
+	if err := r.Client.List(context.TODO(), accountList, listOpts...); err != nil {
+		return nil, err
+	}
+	var accounts []awsv1alpha1.Account
+	for _, account := range accountList.Items {
+		// gets accounts belonging to poo
+		if account.IsOwnedByAccountPool() && account.Spec.AccountPool == accountPoolName {
+			accounts = append(accounts, account)
+		}
+	}
+	return accounts, nil
+
+}
+
+// Updates Account Spec ServiceQuotas to match what's in the ConfigMap
+func (r *AccountPoolValidationReconciler) checkAccountServiceQuota(reqLogger logr.Logger, accountPoolName string, parsedRegionalServiceQuotas awsv1alpha1.RegionalServiceQuotas, accountPoolValidationEnabled bool) (ctrl.Result, error) {
+	accountList, err := r.getAccountPoolAccounts(accountPoolName)
+	if err != nil {
+		reqLogger.Error(err, "Failed to get AccountPool accounts")
+		return reconcile.Result{}, err
+	}
+	var updatedAccountSpecs []awsv1alpha1.Account
+
+	for _, account := range accountList {
+		accountCopy := account
+		if !reflect.DeepEqual(accountCopy.Spec.RegionalServiceQuotas, parsedRegionalServiceQuotas) {
+			accountCopy.Spec.RegionalServiceQuotas = parsedRegionalServiceQuotas
+			if !accountPoolValidationEnabled {
```

Comment:
But this will only print the lines for a single account anyway and wont loop over the accounts, which makes it rather unexpected to implement it in this way. I would've expected this to either print the Expected vs actual Servicequota for all accounts and the exit, or for none.

Reply:
That a good point I'll update that